pontoon/base/views.py

```diff
@@ -76,34 +81,81 @@ def home(request):
     return translate(request, locale, project.slug)
 
 
-def locale(request, locale, template='locale.html'):
-    """Locale view."""
-    log.debug("Locale view.")
+class TemplateModelView(TemplateView):
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(TemplateModelView, self).get(request, *args, **kwargs)
+        except ObjectDoesNotExist:
+            return self.object_does_not_exist(request, *args, **kwargs)
 
-    # Validate locale
-    try:
-        l = Locale.objects.get(code__iexact=locale)
-    except Locale.DoesNotExist:
-        raise Http404
+    def object_does_not_exist(self, *args, **kwargs):
+        return Http404()
 
-    projects = Project.objects.filter(
-        disabled=False, pk__in=Resource.objects.values('project'), locales=l) \
-        .order_by("name")
 
-    if not projects:
-        messages.error(
-            request, "Oops, no projects available for this locale.")
-        request.session['translate_error'] = {
+class ContributorsView(TemplateView):
```

Comment:
I like this class, it's clever that it's both a standalone page and a base class for other pages.